pkg/kubelet/logs/container_log_manager.go

```diff
@@ -213,63 +223,110 @@ func (c *containerLogManager) Clean(ctx context.Context, containerID string) err
 	return nil
 }
 
+func (c *containerLogManager) processQueueItems(ctx context.Context, worker int) {
+	klog.V(4).InfoS("Starting container log rotation worker", "workerID", worker)
+	for c.processContainer(ctx, worker) {
+	}
+	klog.V(4).InfoS("Terminating container log rotation worker", "workerID", worker)
+}
+
 func (c *containerLogManager) rotateLogs(ctx context.Context) error {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
+	klog.V(4).InfoS("Starting container log rotation sequence")
 	// TODO(#59998): Use kubelet pod cache.
 	containers, err := c.runtimeService.ListContainers(ctx, &runtimeapi.ContainerFilter{})
 	if err != nil {
 		return fmt.Errorf("failed to list containers: %v", err)
 	}
-	// NOTE(random-liu): Figure out whether we need to rotate container logs in parallel.
 	for _, container := range containers {
 		// Only rotate logs for running containers. Non-running containers won't
 		// generate new output, it doesn't make sense to keep an empty latest log.
 		if container.GetState() != runtimeapi.ContainerState_CONTAINER_RUNNING {
 			continue
 		}
-		id := container.GetId()
-		// Note that we should not block log rotate for an error of a single container.
-		resp, err := c.runtimeService.ContainerStatus(ctx, id, false)
-		if err != nil {
-			klog.ErrorS(err, "Failed to get container status", "containerID", id)
-			continue
-		}
-		if resp.GetStatus() == nil {
-			klog.ErrorS(err, "Container status is nil", "containerID", id)
-			continue
+		// Doing this to avoid additional overhead with logging of label like arguments that can prove costly
+		if v := klog.V(4); v.Enabled() {
+			klog.V(4).InfoS("Adding new entry to the queue for processing", "id", container.GetId(), "name", container.Metadata.GetName(), "labels", container.GetLabels())
```

Comment:
Do we need to include the labels? Would the pod name/id be more useful instead?

Reply:
The ID and name fields are already present. I will cleanup the labels if you think they are an overkill. The only reason I retained them was because they had a few metadata around the pod directly that can be used for debugging things. But I am ok to get rid of that

Reply:
Container name can be misleading as multiple containers can have the same one. But combined with ID, it is much better. But it is still hard to associate that with a pod. Adding the label makes that slightly easy I guess. What do you think @mrunalp  ?

Reply:
we can do this in a follow up if needed